### PR TITLE
Update department pages and add latest news

### PR DESCRIPTION
**Pull request by mtruj013**
## Done

- Updated department pages and added latest news module
- Included functionality for fast track and featured roles (not all departments are going to have these)

## QA

- Check out this feature branch
- Run the site using the command `./run serve`
- View the site locally in your web browser at: 
   - https://canonical-com-701.demos.haus/careers/engineering
     - doc: https://docs.google.com/document/d/12GWF6eVKezvENc66p3QZ5ZEO8H2QBIlU56RPZNdCPdI/
   - https://canonical-com-701.demos.haus/careers/administration
     - doc: https://docs.google.com/document/d/1Kfkdy1rwBUP1L-ToUVvgqv8WkwEa8SurILkwJJh8u0U/ 
   - https://canonical-com-701.demos.haus/careers/finance
     - doc: https://docs.google.com/document/d/1e6A3z1KsVNyShx2mHq3lNrYkXCaHy8x8BA7qzLxeuJ4/
   - https://canonical-com-701.demos.haus/careers/legal
     - doc: https://docs.google.com/document/d/1TcKYSlfV2G9X_0l-T6rZ-CWd_XJQ_4hHQCPL2CVrCts 
   - https://canonical-com-701.demos.haus/careers/marketing
     - doc: https://docs.google.com/document/d/1larR0nWbVEqeUQbIJdgWemr7I6cqzoUhA9MD6sRQZwE
   - https://canonical-com-701.demos.haus/careers/operations
     - doc: https://docs.google.com/document/d/1zLFku18ZAMVStgkIO3bCoKEXW6ShwMEkB0w9d_nCx9U/
   - https://canonical-com-701.demos.haus/careers/people
     - doc: https://docs.google.com/document/d/1sj-WToRUa9N2c5brs8iAoOfOPHJCeHwvHByq8qcSgK4
   - https://canonical-com-701.demos.haus/careers/project-management
     - doc:  https://docs.google.com/document/d/1hc5olKZcPIHFtxYAHvYFS_c-NZIYy3ZT8WqG7r021MI/
   - https://canonical-com-701.demos.haus/careers/sales
     -  doc: https://docs.google.com/document/d/1yCukmqr7do6iJgl6hKui9lAslWJ6Gol4O2E5ctCE7PY
   - https://canonical-com-701.demos.haus/careers/support-engineering
     - doc:  https://docs.google.com/document/d/1TkePZPzjdIMRTPIep6C1f8SnTPmkKiGdMyUEPMK9rH8
   - https://canonical-com-701.demos.haus/careers/web-and-design
     - doc: https://docs.google.com/document/d/1Eq3USN12tdL1sEf0DMYxO9FAJ5Vqkrl_yeYdRsfWNww/edit
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- Test that links to all roles and individual jobs work as expected

## Issue / Card

Fixes https://warthogs.atlassian.net/browse/WD-536

### CODE REVIEW

**Comment by lyubomir-popov** [comment]
@marinacastejon @mtruj013 Can we do something like this for the featured roles?

image

- rows with `gap: 0`
- cards with padding .5rem 1rem 1rem 1rem;`
- hrs with no `margin-bottom`
- card hover/active states using  these vanilla varialbes for background opacity (color rgba(0,0,0, {insert var here})): 
- 
```
$input-background-opacity-amount: 0.04;
$hover-background-opacity-amount: 0.05;
```

**Comment by marinacastejon** [comment]
Hi @mtruj013 there are a few things to be updated, but you might want to wait for @lyubomir-popov to update the mock-ups. 
- Title "Fast track jobs" should be updated as described in this [JIRA issue](https://warthogs.atlassian.net/browse/WD-535). Lyubo was briefed. 
- From what I see in the mock-ups from visual the photo is not in the right place, it should be in the middle of the content, not just after the heading.  
- In the section with the social media links, the pages where github shows should only show Github's logo, not the branches and the stars. This is an error from the mock-ups, I have tagged Lyubo in figma to fix it. 

Thanks!

File: static/sass/styles.scss [hunk 1]
```diff
@@ -285,3 +285,32 @@ $p-small-lh-diff: map-get($line-heights, default-text) -
...
+  &.is-fast-track &__item {
+     background-color: $color-light;
+  }
+
+  &__item {
+    // @include vf-transition(#{background-color, border-color}, snap, in);
+    
+    margin-right: 1px;
+    border-top: 1px solid $colors--light-theme--border-default;
+    padding: 0 1rem 1rem;
```
Comment (sowasred2012) [inline static/sass/styles.scss:302]: should be alphabetical order

**Review by sowasred2012 — COMMENTED** [review]
A point for design, but it really wasn't clear to me what the link was on these blog sections - I thought it'd be the image initially, then I went looking for a "Read more" esque link, then realised it's the heading: 
image

Also, I don't think we should show the "[...]", it looks messy. Is there a short summary we could pull in instead?

File: static/sass/styles.scss [hunk 2]
```diff
@@ -285,3 +285,32 @@ $p-small-lh-diff: map-get($line-heights, default-text) -
...
+
+.p-featured {
+  gap: 0;
+
+  &.is-fast-track &__item {
+     background-color: $color-light;
+  }
+
+  &__item {
+    // @include vf-transition(#{background-color, border-color}, snap, in);
```
Comment (sowasred2012) [inline static/sass/styles.scss:298]: commented out code

File: templates/careers/administration.html [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      </div>
+    </div>
+  </div>
+  <div class="p-strip is-shallow u-no-padding--top">
+    <div class="u-fixed-width u-hide--small u-hide--medium">
+      {{ image (
+        url="https://assets.ubuntu.com/v1/fcdf52cb-department-admin-%402x.jpg",
+        alt="",
+        width="2216",
+        height="834",
```
Comment (sowasred2012) [inline templates/careers/administration.html:58]: This image can't physically be wider than 1242px, so I think setting these dimensions will reduce the file size when pulling it from cloudinary:
```suggestion
        width="1242",
        height="467",
```

**Review by anthonydillon — APPROVED** [review]
QA looks good to me :+1:

**Review by sowasred2012 — CHANGES_REQUESTED** [review]
Looking good overall, but a few comments in the code - I also think there's some possibilities to move things into partials to cut down the amount of repeated code, or code where the only difference is something minor, like copy or an ID, i.e. the header and footer - these pages will be monsters to update later otherwise. Happy to pair on it.

**Comment by lyubomir-popov** [comment]
@sowasred2012 good point, can we make the blog titles blue?

**Comment by sowasred2012** [comment]
@Zoospora could you give this a design review pls?

**Comment by Zoospora** [comment]
Some comments regarding the featured positions: 
It would be great if we could adjust a little more to lyubo's proposal (a few messages above). I attach again the image, although in the message there are more precise indications about its implementation.

example

I would swear that according to Vanilla the last rows in a table do not usually have a bottom line. 
It would be ideal if the height is consistent so that the sections don't look so close together. 
Even if there is only one element in a second row, I think it would be ideal if the top line went all the way to the end. I am attaching some screenshots of the details mentioned:

finance
772646a999d.png)
operations
16880bb.png)
webanddesign

Thanks!

**Comment by marinacastejon** [comment]
Hi @mtruj013 , 
The [sales page](https://canonical-com-701.demos.haus/careers/sales) doesn't load the blog images for me. 
Other than that, looks good to me, thanks for implementing the changes I mentioned.

**Comment by petesfrench** [comment]
This button at the bottom of the /careers page take you to /careers (ie. back to the top of the same page). Is this the intended behaviour?
image

**Comment by petesfrench** [comment]
Hey @Zoospora, I have made those changes could you take another look please? I wasn't completely sure what you meant by the same height, so I have given each job posting a fixed height.

**Comment by lyubomir-popov** [comment]
> Hey @Zoospora, I have made those changes could you take another look please? I wasn't completely sure what you meant by the same height, so I have given each job posting a fixed height.

I've adjusted them as they should be, not sure fixed hieght is a good idea - sooner or later something will overflow. The height should be the text's height plus a set margin-bottom.

**Comment by lyubomir-popov** [comment]
@petesfrench aontehr thing I noticed - all headings should be `p-heading--2`. There's a few places where there are h3s, h1s, etc: 
image

**Comment by Zoospora** [comment]
> > Hey @Zoospora, I have made those changes could you take another look please? I wasn't completely sure what you meant by the same height, so I have given each job posting a fixed height.
> 
> I've adjusted them as they should be, not sure fixed hieght is a good idea - sooner or later something will overflow. The height should be the text's height plus a set margin-bottom.

perfect, thanks!!

**Comment by petesfrench** [comment]
@lyubomir-popov I have applied the changes from sales onto the rest of the pages and updated the remaining h3's to be p-heading--2. I have updated the scss to not be a fixed height, can you take a look please

**Comment by Zoospora** [comment]
Hey, thanks for the changes! I just see one thing that doesn't quite add up for me. I think the hover should go all the way to the divider line itself and not leave that little margin in between. I'm sorry if I missed any explanation at some point. Thanks in advice!

hover

**Comment by mtruj013** [comment]
> This button at the bottom of the /careers page take you to /careers (ie. back to the top of the same page). Is this the intended behaviour? @marinacastejon image

Hey @petesfrench, thanks for helping with this! Just to add some context, this routes back to the same page because /careers/departments has been deleted and there's a redirect in place that routes the user back to this page if they try to visit the old departments overview. There is a dependant [PR](https://github.com/canonical/canonical.com/pull/720) that deletes this section altogether though. I separated the work for the careers landing page so as to not add to this already massive pr, with the idea that it would be merged directly after this one

**Pull request merged**